deepchecks/nlp/text_data.py

```diff
@@ -65,210 +62,116 @@ class TextData:
-                    if pd.notnull(label).any():
-                        raise DeepchecksValueError('task_type must be set when label is provided')
-                else:
-                    raise DeepchecksValueError('task_type must be set when label is provided')
-
+                raise DeepchecksValueError('task_type must be set when label is provided')
             self._task_type = TaskType.OTHER
         elif task_type == 'text_classification':
             self._task_type = TaskType.TEXT_CLASSIFICATION
         elif task_type == 'token_classification':
+            if tokenized_text is None:
+                raise DeepchecksValueError('tokenized_text must be provided for token_classification task type')
+            validate_tokenized_text(tokenized_text)
+            modified = [[str(token) for token in tokens_per_sample] for tokens_per_sample in tokenized_text]
+            self._tokenized_text = np.asarray(modified, dtype=object)
             self._task_type = TaskType.TOKEN_CLASSIFICATION
         else:
             raise DeepchecksNotSupportedError(f'task_type {task_type} is not supported, must be one of '
                                               f'text_classification, token_classification, other')
 
         if raw_text is None and tokenized_text is None:
-            raise DeepchecksValueError('raw_text and tokenized_text cannot both be None')
+            raise DeepchecksValueError('Either raw_text or tokenized_text must be provided')
         elif raw_text is None:
-            self._validate_tokenized_text(tokenized_text)
-            self._tokenized_text = list(tokenized_text)
-            self._text = [' '.join(tokens) for tokens in tokenized_text]
-        elif tokenized_text is None:
-            self._validate_text(raw_text)
-            self._text = list(raw_text)
-            if self._task_type == TaskType.TOKEN_CLASSIFICATION:
-                self._tokenized_text = [sample.split() for sample in self._text]
-            else:
-                self._tokenized_text = None
-        else:
-            self._validate_text(raw_text)
-            self._validate_tokenized_text(tokenized_text)
-            self._text, self._tokenized_text = list(raw_text), list(tokenized_text)
-            if len(raw_text) != len(tokenized_text):
-                raise DeepchecksValueError('raw_text and tokenized_text must have the same length')
-
-        if index is None:
-            self.index = list(range(len(raw_text)))
-        elif len(index) != len(raw_text):
-            raise DeepchecksValueError('index must be the same length as raw_text')
+            self._text = np.asarray([' '.join(tokens) for tokens in tokenized_text])  # Revisit this decision
         else:
-            self.index = list(index)
+            validate_raw_text(raw_text)
+            self._text = np.asarray([str(x) for x in raw_text])
+            if tokenized_text is not None and len(raw_text) != len(tokenized_text):
```

Comment:
"if tokenized_text is not None" not necessary - you're in the "else" block anyway, it must mean raw_text=None and tokenized_text!=None

Reply:
its raw_text!=None and tokenized text can be both

Reply:
Yeah, sorry, misread that